packages/opentelemetry-plugin-grpc/src/grpc.ts

```diff
@@ -127,22 +123,9 @@ export class GrpcPlugin extends BasePlugin<grpc> {
     }
   }
 
-  private _getSpanContext(
-    metadata: grpcTypes.Metadata
-  ): SpanContext | undefined {
-    return getExtractedSpanContext(
-      this._tracer.getHttpTextFormat().extract(Context.TODO, metadata.getMap())
-    );
-  }
-
-  private _setSpanContext(
-    metadata: grpcTypes.Metadata,
-    spanContext: SpanContext
-  ): void {
+  private _setSpanContext(metadata: grpcTypes.Metadata): void {
     const carrier = {};
-    this._tracer
-      .getHttpTextFormat()
-      .inject(setExtractedSpanContext(Context.TODO, spanContext), carrier);
+    propagation.inject(carrier);
```

Comment:
default is `NOOP_HTTP_TEXT_FORMAT`, where do we init `HttpTraceContext` for grpc plugin?

Reply:
Ohh it is defined in the global API, right? Which means all the plugins will get same propagator?

Reply:
What if I want `BinaryFormat` for grpc and `HttpTraceContext` for http? How do you do that?

Reply:
Ignore earlier comment, just read about **Composite Propagator**.